pkg/auth/service.go

```diff
@@ -1040,19 +1011,39 @@ func (s *DBAuthService) ClaimTokenIDOnce(ctx context.Context, tokenID string, ex
 }
 
 // markTokenSingleUse returns true if token is valid for single use
-func (s *DBAuthService) markTokenSingleUse(ctx context.Context, tokenID string, tokenExpiresAt time.Time) (bool, error) {
-	res, err := s.db.Exec(ctx, `INSERT INTO auth_expired_tokens (token_id, token_expires_at) VALUES ($1,$2) ON CONFLICT DO NOTHING`,
-		tokenID, tokenExpiresAt)
+func (s *KVAuthService) markTokenSingleUse(ctx context.Context, tokenID string, tokenExpiresAt time.Time) (bool, error) {
+	tokenPath := model.KvTokenPath(tokenID)
+	m := model.TokenData{TokenId: tokenID, ExpiredAt: timestamppb.New(tokenExpiresAt)}
+	err := s.store.SetMsg(ctx, model.PartitionKey, tokenPath, &m)
 	if err != nil {
 		return false, err
 	}
-	canUseToken := res.RowsAffected() == 1
+
 	// cleanup old tokens
-	_, err = s.db.Exec(ctx, `DELETE FROM auth_expired_tokens WHERE token_expires_at < $1`, time.Now())
+	t := model.TokenData{}
+	itr, err := s.store.Scan(ctx, t.ProtoReflect().Type(), model.PartitionKey, model.KvTokenPath(""))
 	if err != nil {
-		s.log.WithError(err).Error("delete expired tokens")
+		return false, fmt.Errorf("sacn tokens: %w", err)
+	}
+	defer itr.Close()
+
+	for itr.Next() {
+		entry := itr.Entry()
+		token, ok := entry.Value.(*model.TokenData)
+		if !ok {
+			s.log.WithError(err).Error("delete expired tokens")
+		}
+		if token.ExpiredAt.AsTime().Before(time.Now()) {
+			err = s.store.DeleteMsg(ctx, model.PartitionKey, model.KvTokenPath(token.TokenId))
+			if err != nil {
+				s.log.WithError(err).Error("delete expired tokens")
+			}
+		}
+	}
+	if err = itr.Err(); err != nil {
+		return true, err
 	}
-	return canUseToken, nil
+	return true, nil
```

Comment:
Is it always true?

Reply:
It is true if the save of the token in the DB succeeded